backend/src/main/java/de/neuefische/booklibrary/backend/SecurityConfig.java

```diff
@@ -0,0 +1,113 @@
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.UserDetailsManager;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+
+@EnableWebSecurity
+@RequiredArgsConstructor
+public class SecurityConfig {
+    private static final String EXCEPTION_MSG = "You cannot use this custom UserDetailsManager for this action.";
+    private static final String LIBRARIAN_ROLE = "LIBRARIAN";
+
+    private final AppUserService appUserService;
+
+    public static final PasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+
+    @Bean
+    public PasswordEncoder encoder() {
+        return passwordEncoder;
+    }
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        return http
+                .csrf().csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse())
+                .and()
+                .httpBasic()
+                .and()
+                .authorizeRequests()
+
+                .antMatchers(HttpMethod.GET, "/api/app-users/login").permitAll()
+                .antMatchers(HttpMethod.GET, "/api/app-users/me").permitAll()
+                .antMatchers(HttpMethod.GET, "/api/app-users/user").permitAll()
+                .antMatchers(HttpMethod.GET, "/api/app-users/role").authenticated()
+                .antMatchers(HttpMethod.GET, "/api/app-users/logout").authenticated()
+                .antMatchers(HttpMethod.POST, "/api/app-users/member").permitAll()
+                .antMatchers(HttpMethod.POST, "/api/app-users/librarian").permitAll()
+                .antMatchers(HttpMethod.DELETE, "/api/app-users/deleteMe").authenticated()
+
+
+                .antMatchers(HttpMethod.GET, "/api/books").permitAll()
+                .antMatchers(HttpMethod.GET, "/api/books/rentedBooks").permitAll()
+                .antMatchers(HttpMethod.GET, "/api/books/search/**").hasRole(LIBRARIAN_ROLE)
+                .antMatchers(HttpMethod.GET, "/api/books/isbn/**").hasRole(LIBRARIAN_ROLE)
+                .antMatchers(HttpMethod.POST, "/api/books").hasRole(LIBRARIAN_ROLE)
+                .antMatchers(HttpMethod.POST, "api/books/rentBook/**").hasRole(LIBRARIAN_ROLE)
+                .antMatchers(HttpMethod.POST, "api/books/returnBook/**").hasRole(LIBRARIAN_ROLE)
```

Comment:
Soweit ich mich erinnere müssen URLs in der Security-Config mit einem `/` starten. Bitte kontrolliere das nochmal.
```suggestion
                .antMatchers(HttpMethod.POST, "/api/books/rentBook/**").hasRole(LIBRARIAN_ROLE)
                .antMatchers(HttpMethod.POST, "/api/books/returnBook/**").hasRole(LIBRARIAN_ROLE)
```

Reply:
Gut gesehen! Die URLs sind angepasst